src/cast/safe/grow.rs

```diff
@@ -0,0 +1,113 @@
+/// Cast from a smaller number to a larger one without changing sign.
+///
+/// The `GrowFrom<T>` trait is similar to `std::convert::From<T>`. However,
+/// while `std::convert::From<T>` performs a logical conversion, `GrowFrom<T>`
+/// performs a bitwise cast from a number to a number of the same sign but
+/// of a possibly larger size. `GrowFrom<T>` will **never** decrease the size
+/// of a number or change from an integer of one signedness to the other.
```

Comment:
Calling this "bitwise" is again weird.  Otherwise, the difference from `From` is just that you're preserving sign, but I wonder how useful that really is in generic code?  If you care to remain signed, you can add a constraint with the `Signed` trait, or likewise with the `Unsigned` trait.  But if you're in generic code that doesn't know whether it's signed or unsigned, I'm not sure that I see it being useful to assert that the signedness won't change.

Reply:
That is true, but that behavior is implicit. A big part of these traits is turning implicit behavior into explicit behavior. This improves readability and reduces run time errors.